controllers/post.js

```diff
@@ -3,19 +3,42 @@ const express = require("express");
 const Users = require("../models/users");
 const Groups = require("../models/groups");
 const Posts = require("../models/posts");
+const PostsLikes = require("../models/posts_likes");
 
 const router = express.Router();
 
 const validation = require("../middleware/dataValidation");
 
 const { postSchema } = require("../schemas");
 
+async function addLikesToPost(posts, error, res) {
```

Comment:
It's not hurting anything by being in this controller file, but from a separation of concerns standpoint this could be extracted into a file in the middleware folder and imported instead